libs/components/core/src/lib/modules/log/log.service.spec.ts

```diff
@@ -1,25 +1,465 @@
-import { TestBed, inject } from '@angular/core/testing';
+import { TestBed } from '@angular/core/testing';
 
 import { SkyLogService } from './log.service';
+import { SkyLogLevel } from './types/log-level';
+import { SKY_LOG_LEVEL } from './types/log-level-token';
 
 describe('Log service', () => {
+  let consoleDebugSpy: jasmine.Spy;
+  let consoleErrorSpy: jasmine.Spy;
+  let consoleInfoSpy: jasmine.Spy;
+  let consoleLogSpy: jasmine.Spy;
+  let consoleWarnSpy: jasmine.Spy;
+  let logService: SkyLogService;
+
+  function setupSpys(): void {
+    consoleDebugSpy = spyOn(console, 'debug');
+    consoleErrorSpy = spyOn(console, 'error');
+    consoleInfoSpy = spyOn(console, 'info');
+    consoleLogSpy = spyOn(console, 'log');
+    consoleWarnSpy = spyOn(console, 'warn');
```

Comment:
**Non-blocking feedback:** you could use `jasmine.createSpyObj` instead, but what you have now is much better than what it was before. 👍 

```
let consoleSpy: jasmine.SpyObj<Console>;

beforeEach(() => {
  consoleSpy = jasmine.createSpyObj('console', [
    'debug',
    'error',
    'info',
    'log',
    'warn'
  ]);
});

it('should', () => {
  // ...
  expect(consoleSpy.debug).not.toHaveBeenCalled();
  expect(consoleSpy.error).toHaveBeenCalledWith('Test');
  expect(consoleSpy.info).not.toHaveBeenCalled();
  expect(consoleSpy.log).not.toHaveBeenCalled();
  expect(consoleSpy.warn).not.toHaveBeenCalled();
});
```

Reply:
Done